modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1140,7 +1140,7 @@ bool CvCapture_FFMPEG::open(const char* _filename, const VideoCaptureParameters&
         CV_WARN(_filename);
         goto exit_func;
     }
-    err = avformat_find_stream_info(ic, NULL);
+    err = avformat_find_stream_info(ic, &dict);
```

Comment:
Currently `dict` is targeted to `avformat_open_input()` call above.

Reusing it with `avformat_find_stream_info()` may provide conflicts or unexpected behavior.
This should be checked first.

Reply:
Can you think of a way to determine that except reading all the source? Afaics it simply extends the number of available options to those provided by muxers/demuxers and protocols. 
Additionally passing it to avcodec_open2 would widen the available options to those provided by encoders/decoders.

Reply:
Adding dedicated variable `OPENCV_FFMPEG_CAPTURE_OPTIONS_STREAM` should avoid any possible conflicts.

---

**Update**:

Looks like `avformat_find_stream_info` didn't read passed dictionary.
It writes into it.

Ref: https://ffmpeg.org/doxygen/trunk/group__lavf__decoding.html#gad42172e27cddafb81096939783b157bb

> **options**	If non-NULL, an ic.nb_streams long array of pointers to dictionaries, where i-th member contains options for codec corresponding to i-th stream. On return each dictionary will be filled with options that were not found.

Moreover it need not a single dictionary. It wants an **array**.

---

So topic with usage scenario is still open.

Reply:
> Adding dedicated variable `OPENCV_FFMPEG_CAPTURE_OPTIONS_STREAM` should avoid any possible conflicts.

Very good idea!

> 
> **Update**:
> 
> Looks like `avformat_find_stream_info` didn't read passed dictionary. It writes into it.

That is not the whole truth.. for every stream there is a call to ``if (avcodec_open2(avctx, codec, options ? &options[i] : &thread_opt) < 0)``` which in turn reads from the dictionary.

> 
> Moreover it need not a single dictionary. It wants an **array**.

And that it gets (&dict with dict beeing a pointer).

> 
> So topic with usage scenario is still open.

You mean this https://github.com/opencv/opencv/pull/22708#issuecomment-1294827715 isn't clear enough? What should i clarify?

Reply:
> &dict with dict beeing a pointer

It is a pointer to **single** element.

Passing `&options[i]` will lead to crash or memory corruption for `i >= 1` values

Reply:
I guess you are right though I still try to wrap my head all around the ffmpeg code :). I'll get back to it tomorrow and see how this might make sense.
Thank you!

Reply:
> Passing `&options[i]` will lead to crash or memory corruption for `i >= 1` values

Definitely true. I still have to figure out how to do this the right way. Anyway when i find out I'll implement it using ```OPENCV_FFMPEG_CAPTURE_OPTIONS_STREAM```

Reply:
This needs a way more comprehensive approach then just passing a dict. I think an approach like [ffplay.c](https://ffmpeg.org/doxygen/trunk/ffplay_8c-source.html) does using [cmdutils.h](https://ffmpeg.org/doxygen/0.6/cmdutils_8h.html)

My idea is to support filter graph parsing like it is supported for gstreamer.
Example:

```c++
VideoCapture("-v trace -f v4l2src -input_format mjpeg -i /dev/video0 -vf scale=320:240", CAP_FFMPEG) cap;
```
What do you think?

Reply:
Anyway. this pull request and the pertaining issue can be discarded as invalid and I would create a new issue

Reply:
In the meantime I implemented a compromise: 198c0b8b0d45b73e41222f4276ccd15bf70a18c5